pkg/model/components/kubedns.go

```diff
@@ -74,5 +74,29 @@ func (b *KubeDnsOptionsBuilder) BuildOptions(o interface{}) error {
 		clusterSpec.KubeDNS.MemoryLimit = &defaultMemoryLimit
 	}
 
+	NodeLocalDNS := clusterSpec.KubeDNS.NodeLocalDNS
+	if NodeLocalDNS == nil {
+		NodeLocalDNS.Disable = true
+	} else if !NodeLocalDNS.Disable {
+		// https://kubernetes.io/docs/tasks/administer-cluster/nodelocaldns/#configuration
+		NodeLocalDNS.Domain = clusterSpec.ClusterDNSDomain
+		NodeLocalDNS.LocalIP = "169.254.20.10"
```

Comment:
Not sure here about this. Just copying here from the official doc.

Also I would need to implement this as the official doc states:

> If using kube-proxy in IPVS mode, --cluster-dns flag to kubelet needs to be modified to use <node-local-address> that NodeLocal DNSCache is listening on. Otherwise, there is no need to modify the value of the --cluster-dns flag, since NodeLocal DNSCache listens on both the kube-dns service IP as well as <node-local-address>.

Reply:
Added in this commit https://github.com/kubernetes/kops/pull/8780/commits/b0cb7c1795a76bb1bd5d44542bdb11dae12ec92f

Reply:
First this is made configurable by adding it to the cluster API and then it is just hardcoded to the address above unconditionally?
Isn't it better to test for this value in the template and set it to the address above if the value is empty?